### PR TITLE
added support for SASS_PATH env variable

### LINKED ISSUE #253: SASS_PATH is ignored
Ruby version of Sass documents environment variable `SASS_PATH` to used to add  load paths. sassc doesn't seem to. This feature is very usefull when Sass files come from different sources and   can be located in different directories and paths to those directories may vary depending on situation. In particular, if the files are pulled by a package manager.

### PR DESCRIPTION
**Pull request by grisumbras**
Fixes #253

### CODE REVIEW

File: sassc.c [hunk 1]
```diff
@@ -231,6 +231,28 @@ void invalid_usage(char* argv0) {
...
+    } else {
+        buffer = existing_buffer;
+    }
+
+    memcpy(buffer, begin, count);
+    buffer[count] = '\0';
+    sass_option_push_include_path(options, buffer);
+
+    if (buffer != existing_buffer) {
+        free(buffer);
```
Comment (am11) [inline sassc.c]: This is not required; libsass frees it on deleting the compilation context.

Comment (grisumbras) [reply sassc.c]: I looked into `sass_option_push_include_path` implementation. It copies the contents of the string it was passed and then `options` object owns the copy. The source's ownershup is left to the caller, so the temporary buffer has to be deallocated.

Comment (am11) [reply sassc.c]: Ah, you are right; i forgot about `sass_copy_c_string`.

**Comment by mgreter** [comment]
Couldn't you just use [`sass_option_set_include_path`][https://github.com/sass/libsass/blob/master/docs/api-context.md]. I know it could use some more documentation on the wiki, but it states on https://github.com/sass/libsass/blob/master/docs/api-context-internal.md:
```
  // Colon-separated list of paths
  // Semicolon-separated on Windows
  // Maybe use array interface instead?
  char* include_path;
  char* plugin_path;
```

AFAICT you could also just push the delimited string via `sass_option_push_include_path`, since both version got through [`collect_include_paths`][1], which does the actual [path splitting][2] according to OS dir delimiter.

[1]: https://github.com/sass/libsass/blob/master/src/context.cpp#L81-L82
[2]: https://github.com/sass/libsass/blob/master/src/context.cpp#L151-L182
collect_include_paths

**Comment by mgreter** [comment]
@grisumbras sorry that we didn't make it more obvious, but IMO the only change needed to support this is in https://github.com/sass/sassc/pull/255. Tried it locally and seems to work. Could you double check it?

Thank you very much for your effort, not sure why we didn't spot this earlier, given that I implemented the API on the libsass side exactly with this kind of use case in mind.

**Comment by grisumbras** [comment]
Rewritten to using just a single push.

**Comment by mgreter** [comment]
I would stick to `sass_option_set_include_path`, as it is the documented way to pass a delimited string for now (the other one is undocumented). It also makes the check obsolete, as it will handle `\0` just fine.

`sass_option_set_include_path(options, getenv("SASS_PATH"));`

To elaborate a bit more, `sass_option_set_include_path` was always meant for exactly this use case, while `sass_option_push_include_path` is more intended for tools that support multiple `-I` arguments.

**Comment by grisumbras** [comment]
I changed the implementation to use `sass_option_set_include_path`. Is there anything else that needs to be done?

**Comment by mgreter** [comment]
Could you squash all commits down to one and force update your branch?
After that I'm happy to merge. Thank you!

**Comment by grisumbras** [comment]
Done.

**Comment by mgreter** [comment]
Thank you very much for your contribution.

**Pull request merged**